### PR TITLE
Fix dependencies to allow Dependabot to raise a Rails 6.1 PR

### PR DESCRIPTION
**Pull request by brucebolt**
Dependabot did not raise a Rails 6.1 update PR as there were incompatible dependencies.

- Bumping `inherited_resources` to the latest version to change the requirement for `actionpack (>= 5.0, < 6.1)` to `actionpack (>= 5.2, < 6.2)`
- Bumping `ruby2_keywords` to meet `mongoid` 7.3.2's requirement for `ruby2_keywords (~> 0.0.5)`
- Bumping `mongoid` to the latest version to remove the requirement for `activemodel (>= 5.1, < 6.1)` to `activemodel (>= 5.1, < 6.2)`
- Bumping `select2-rails` to the latest patch of version 3.5 to remove the requirement for `thor (~> 0.14)`, which has required updates to the tests as the `select` element is explicitly marked as `display: none` (so not matched with capybara's `select` method)

Also adding `inherited_resources` and `mongoid` to the dependabot configuration, so we don't get into this situation with the next Rails update.  We don't need to add `ruby2_keywords`, as this is a `mongoid` dependency, so will get bumped alongside this gem, as required.  `select2-rails` doesn't need adding as the `thor` dependency is no longer fixed to a specific version.

[Trello card](https://trello.com/c/V56Gndqj)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -202,10 +202,10 @@ GEM
       domain_name (~> 0.5)
     i18n (1.8.10)
       concurrent-ruby (~> 1.0)
-    inherited_resources (1.11.0)
-      actionpack (>= 5.0, < 6.1)
+    inherited_resources (1.13.0)
```
Comment (ChrisBAshton) [inline Gemfile.lock:206]: Worth explaining in the commit message itself:

```
Bumping inherited_resources to the latest version to change the requirement for actionpack (>= 5.0, < 6.1) to actionpack (>= 5.2, < 6.2)
```

File: Gemfile.lock [hunk 2]
```diff
@@ -277,11 +277,11 @@ GEM
     mocha (1.11.2)
     momentjs-rails (2.20.1)
       railties (>= 3.1)
-    mongo (2.13.0)
+    mongo (2.15.0)
       bson (>= 4.8.2, < 5.0.0)
-    mongoid (7.1.2)
-      activemodel (>= 5.1, < 6.1)
-      mongo (>= 2.7.0, < 3.0.0)
+    mongoid (7.3.1)
```
Comment (ChrisBAshton) [inline Gemfile.lock]: Worth explaining in the commit message itself:

```
Bumping mongoid to the latest version to remove the requirement for activemodel (>= 5.1, < 6.1) to activemodel (>= 5.1, < 6.2)
```

File: Gemfile.lock [hunk 3]
```diff
@@ -418,7 +418,7 @@ GEM
     ruby-graphviz (1.2.5)
       rexml
     ruby-progressbar (1.11.0)
-    ruby2_keywords (0.0.4)
+    ruby2_keywords (0.0.5)
```
Comment (ChrisBAshton) [inline Gemfile.lock:423]: Worth explaining in the commit message itself:

```
Bumping ruby2_keywords to meet mongoid 7.3.2's requirement for ruby2_keywords (~> 0.0.5)
```

File: Gemfile [hunk 4]
```diff
@@ -39,6 +39,7 @@ gem "uglifier"
 gem "whenever", require: false
 
 group :test do
+  gem "capybara-select-2"
```
Comment (ChrisBAshton) [inline Gemfile:42]: The link in the commit message leads to a "There is nothing to compare" page 🤔  https://github.com/argerim/select2-rails/compare/v3.5.9.1...v3.5.10diff-ffea88f08f5750e44aab693b050ad1f4e03fc54ec95a48f814dd67395ec0df95R886

**Review by ChrisBAshton — APPROVED** [review]
Excellent job getting to the bottom of this! 👏 

Being nitpicky, some of the PR description should ideally be copied into the respective commits. There also seems to be a broken link in one of the commits that ought to be fixed. Happy to approve and leave these with you though.

File: Gemfile (hunk 4 above)
Comment (brucebolt) [reply Gemfile:42]: It looks like GitHub removes the hash symbol from the URL when rendering the commit message in the UI.

The [full URL](https://github.com/argerim/select2-rails/compare/v3.5.9.1...v3.5.10#diff-ffea88f08f5750e44aab693b050ad1f4e03fc54ec95a48f814dd67395ec0df95R886) is there when the repo is pulled and the git log inspected.

Comment (ChrisBAshton) [reply Gemfile:42]: Ahhh, interesting! Sounds like a [ways to improve GitHub](https://twitter.com/search?q=Ways%20to%20improve%20%40github%20(from%3Achrisbashton)&src=typed_query&f=live) suggestion to me...! 🤔

**Comment by brucebolt** [comment]
@ChrisBAshton I've added a couple of commits ([1](https://github.com/alphagov/publisher/pull/1489/commits/65922fcaa93059c51197f9377a88821c11afd144)], [2](https://github.com/alphagov/publisher/pull/1489/commits/1c405179d6d4f023bb76819f09ffc0cc6ae10c1c) to resolve issues with changes in https://github.com/alphagov/publisher/pull/1478.

Would you be able to review these two commits again?  Nothing else has changed in the other commits, other than the commit message.

File: app/views/editions/_reviewer_field.html.erb [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<%= form_group(f, :reviewer) do %>
+<%= form_group(f, :reviewer, label: "Reviewer") do %>
```
Comment (ChrisBAshton) [inline app/views/editions/_reviewer_field.html.erb:1]: Does this conflict with our translations work? Should this refer to a i18n yaml property instead?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍  one possibly blocking comment below.

File: app/views/editions/_reviewer_field.html.erb (hunk 5 above)
Comment (brucebolt) [reply app/views/editions/_reviewer_field.html.erb:1]: The translation work is only being done on the frontend apps, so publisher isn't affected.  If we decide to translate publisher, then we'll need to audit all the hardcoded strings (like we did in the frontend apps).

**Comment by brucebolt** [comment]
This has allowed dependabot to raise https://github.com/alphagov/publisher/pull/1493.

**Pull request merged**